Sources/Sentry/Transactions/SentryTransaction.m

```diff
@@ -0,0 +1,118 @@
+//
+//  SentryTransaction.m
+//  Sentry
+//
+//  Created by Dhiogo Brustolin on 11/01/21.
+//  Copyright © 2021 Sentry. All rights reserved.
+//
+
+#import "SentryTransaction.h"
+#import "NSDate+SentryExtras.h"
+#import "NSDictionary+SentrySanitize.h"
+#import "SentryCurrentDate.h"
+#import "SentryHub.h"
+#import "SentryId.h"
+#import "SentrySpanContext.h"
+#import "SentrySpanId.h"
+#import "SentryTransactionContext.h"
+
+@interface
+SentryTransaction () {
+    SentrySpanContext *trace;
+}
+
+@end
+
+@implementation SentryTransaction
+
+- (NSDictionary<NSString *, id> *)serialize
+{
+    if (nil == self.timestamp) {
+        self.timestamp = [SentryCurrentDate date];
+    }
+
+    NSMutableDictionary *serializedData = @{
+        @"event_id" : self.eventId.sentryIdString,
+        @"timestamp" : [self.timestamp sentry_toIso8601String],
+        @"spans" : @[],
```

Comment:
Why is this empty?

Reply:
Bruno wanted to see some progress so I created this draft PR. Still work to do before consider it done.